api/profile/profileModel.js

```diff
@@ -41,7 +41,9 @@ const remove = (id) => {
 };
 
 const findOrCreateProfile = async (profileObj) => {
-  const foundProfile = await findById(profileObj.id).then((profile) => profile);
+  const foundProfile = await findById(profileObj.profile_id).then(
```

Comment:
From a styling perspective, consider moving your .then() to the next line. It thereby allows the reader to know that we're dealing with a promise situation without having to read through the entire line, first. This can be very helpful when somebody is simply trying to get a general idea of what the code does, and is even more helpful when trying to follow the flow of the logic.